nm ovs: Fix device reapply

cathay4t opened the pull request:
When applying the identical YAML with OVS again, nmstate got failure
when device reapply:

    Reapply operation failed on ovs-interface ovs0
    93f1914e-67ea-4f91-a228-7ae384d285b3, reason:
    IncompatibleReapply:Can't reapply any changes to 'ovs-interface'
    setting, retry on normal activation

The root cause is OVS port and OVS interface are using the same
`connection.interface-name`, but `NmApi::connection_reapply` only search
device using interface name, which lead to nmstate asking reapply
on the wrong NmDevice.

To fix the problem, for OVS connections, we search full list of
NmDevice by matching both interface name and type.

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-50556

cathay4t commented:
Scratch build could be found at: https://people.redhat.com/fge/RHEL-50556/nmstate-2.2.34-0.20240801.2359gitea91aee2.el9.x86_64.rpm

liangwen12year commented:
I think that the integration test `test_preserve_unmanaged_routes` is broken by the current fix, can you check it out ?

liangwen12year commented on rust/src/lib/nm/query_apply/profile.rs:
```diff
@@ -192,7 +192,7 @@ fn _activate_nm_profiles(
         if let Some(uuid) = nm_conn.uuid() {
             log::info!(
-                "Activating connection {}: {}/{}",
+                "Reapplying connection {}: {}/{}",
```
What is the reason changing to `Reapplying connection` here ?

cathay4t replied:
To distinguish activation vs  reapply.

cathay4t commented:
CI fixed.

cathay4t committed:
nm ovs: Fix device reapply

When applying the identical YAML with OVS again, nmstate got failure
when device reapply:

    Reapply operation failed on ovs-interface ovs0
    93f1914e-67ea-4f91-a228-7ae384d285b3, reason:
    IncompatibleReapply:Can't reapply any changes to 'ovs-interface'
    setting, retry on normal activation

The root cause is OVS port and OVS interface are using the same
`connection.interface-name`, but `NmApi::connection_reapply` only search
device using interface name, which lead to nmstate asking reapply
on the wrong NmDevice.

To fix the problem, for OVS connections, we search full list of
NmDevice by matching both interface name and type.

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-50556

Signed-off-by: Gris Ge <fge@redhat.com>
